src/dragonball/src/device_manager/console_manager.rs

```diff
@@ -0,0 +1,427 @@
+                        .set_output_stream(None);
+                }
+                Ok(count) => {
+                    self.device
+                        .lock()
+                        .expect("console: poisoned console lock")
+                        .raw_input(&out[..count])?;
+                    should_drop = false;
+                }
+                Err(e) => {
+                    slog::warn!(self.logger,
+                        "error while reading serial conn sock: {:?}", e;
+                        "subsystem" => "console_mgr"
+                    );
+                    self.device
+                        .lock()
+                        .expect("console: poisoned console lock")
+                        .set_output_stream(None);
+                }
+            }
+        }
+
+        if should_drop {
+            assert!(self.sock_conn.is_some());
+            // Safe to unwrap() because self.sock_conn is Some().
+            let sock_conn = self.sock_conn.take().unwrap();
+            let events = Events::with_data(&sock_conn, EPOLL_EVENT_SERIAL_DATA, EventSet::IN);
+            if let Err(e) = ops.remove(events) {
+                slog::error!(self.logger,
+                    "failed deregister epoll event for UDS, {:?}", e;
+                    "subsystem" => "console_mgr"
+                );
+            }
+        }
+
+        Ok(())
+    }
+
+    fn stdio_read_in(&mut self, ops: &mut EventOps) -> std::io::Result<()> {
+        let mut should_drop = true;
+
+        if self.stdin_handle.is_some() {
+            let mut out = [0u8; 64];
+            // Safe to unwrap() because self.stdin_handle is Some().
+            let stdin_lock = self.stdin_handle.as_ref().unwrap().lock();
+            match stdin_lock.read_raw(&mut out[..]) {
+                Ok(0) => {
+                    // Zero-length read indicates EOF. Remove from pollables.
+                    self.device
+                        .lock()
+                        .expect("console: poisoned console lock")
+                        .set_output_stream(None);
+                }
+                Ok(count) => {
+                    self.device
+                        .lock()
+                        .expect("console: poisoned console lock")
+                        .raw_input(&out[..count])?;
+                    should_drop = false;
+                }
+                Err(e) => {
+                    slog::warn!(self.logger,
+                        "error while reading stdin: {:?}", e;
+                        "subsystem" => "console_mgr"
+                    );
+                    self.device
+                        .lock()
+                        .expect("console: poisoned console lock")
+                        .set_output_stream(None);
+                }
+            }
+        }
+
+        if should_drop {
+            let events = Events::with_data_raw(libc::STDIN_FILENO, EPOLL_EVENT_STDIN, EventSet::IN);
```

Comment:
Is it need to set  `self.stdin_handle` to `None`?

Reply:
If self.stdin_handle is set to None, stdin related connection will be lost. We only want to drop the data in stdin channel here.